src/test/java/uk/gov/hmcts/reform/blobrouter/tasks/SendDailyReportTaskTest.java

```diff
@@ -0,0 +1,141 @@
+package uk.gov.hmcts.reform.blobrouter.tasks;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.hmcts.reform.blobrouter.model.out.EnvelopeSummaryItem;
+import uk.gov.hmcts.reform.blobrouter.services.email.EmailSender;
+import uk.gov.hmcts.reform.blobrouter.services.report.ReportCsvWriter;
+import uk.gov.hmcts.reform.blobrouter.services.report.ReportService;
+
+import java.io.File;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+
+@ExtendWith(MockitoExtension.class)
+class SendDailyReportTaskTest {
+    private SendDailyReportTask sendDailyReportTask;
+
+    @Mock
+    private ReportService reportService;
+
+    @Mock
+    private ReportCsvWriter reportCsvWriter;
+
+    @Mock
+    private EmailSender emailSender;
+
+    private static final String FROM = "From";
+
+    private String[] recipients = new String[]{"rec1", "rec2"};
+
+    @Mock
+    private List<EnvelopeSummaryItem> report;
+
+    @Mock
+    private File reportFile;
+
+    @Captor
+    private ArgumentCaptor<String> subjectCaptor;
+
+    @Captor
+    private ArgumentCaptor<String> bodyCaptor;
+
+    @Captor
+    private ArgumentCaptor<String> fromCaptor;
+
+    @Captor
+    private ArgumentCaptor<String[]> recipientsCaptor;
+
+    @Captor
+    private ArgumentCaptor<Map<String, File>> attachmentsCaptor;
+
+    @BeforeEach
+    void setUp() {
+        sendDailyReportTask = new SendDailyReportTask(
```

Comment:
A suggestion: instead of creating an instance of this task in the `setUp` method, we could just call a method creating the task from test. That method, say `newSendDailyReportTask(...)` could take `from` and `recipients` as parameters. This way we could get rid of those two class-global fields and put them where they're used.